ASTInfoGenerator/DeclParser/DeclParser.py

```diff
@@ -0,0 +1,382 @@
+
+        if not specifier_manager.struct_exists(struct_name):
+            logout(
+                f"did not find for parsed var_decl var name = {var_name} struct name = '{struct_name}'")
+            return None
+
+        logout(var_decl)
+        logout(
+            f"did parsed var_decl var name = {var_name} struct name = '{struct_name}'")
+
+        return StructVarDecl(var_name, struct_name)
+
+    def _parse_anno(self, anno_decl: str, specifier_manager: SpecifierManager, param_index: int, field_index: int):
+        spec = specifier_manager.get_specifiers()[-1]
+
+        logout(f"pre is {anno_decl}")
+
+        start_anno_index = anno_decl.find('"')
+        end_anno_index = anno_decl.rfind('"')
+
+        anno_decl = anno_decl[start_anno_index + 1: end_anno_index]
+
+        anno_type = anno_decl[
+            0: anno_decl.find(" ")
+        ]
+
+        anno_methods = anno_decl[anno_decl.rfind(
+            "methods = ") + len("methods = ")::]
+        logout(f"ANNO METHODS = {anno_methods}")
+
+        known_target = ""
+
+        anno_unfilled_target = ""
+        anno_unfilled_target_index = anno_decl.find("_")
+
+        while (anno_decl[anno_unfilled_target_index] != " "):
+            anno_unfilled_target += anno_decl[anno_unfilled_target_index]
+            anno_unfilled_target_index += 1
+
+        # anno_unfilled_target looks like "_" or "_.FIELD(x)"
+
+        if type(spec) is Struct:
+            known_target = f"STRUCT({spec.get_name()}).FIELD({field_index})"
+
+        elif type(spec) is Function:
+            known_target = f"FUNCTION({spec.get_name()})"
+
+            # Assumption: if it is not referring to a parameter, it is
+            # referring to the function itself
+            if param_index is None:
+                known_target += ".RETURN"
+            else:
+                known_target += f".PARAM({param_index})"
+
+        if ".FIELD" in anno_unfilled_target:
+            anno_unfilled_target_field_name = anno_unfilled_target[
+                anno_unfilled_target.find("(") + 1: anno_unfilled_target.find(")")
+            ]
+
+            if type(spec) is Function:
+                if param_index is None:
+                    logout(f"ret type is '{spec.get_return_type()}'")
+                    # Assumption: spec.return_type looks like 'struct <struct_name> ' or '<struct_name> '
+                    # note: it'll look like '<struct_name> ' if a typedef is used
+
+                    # we can make this assumption because:
+                    # 1. param_index is none, which means we are making an
+                    # annotation for the function itself, and (for now) we assume
+                    # that means it's an annotation on its return.
+                    # note: we'll need to update how we treat annotations
+                    # applied to the function itself
+                    # if/when annotations like @Owning and @MustCallAlias are implemented
```

Comment:
Will we? Both of these examples can be viewed as annotating the return type, too.

The only case that would be difficult is annotations that are explicitly about the function and not about the return; I don't think the RLC uses any of those besides `@EnsuresCalledMethods`.

Reply:
I feel like we had a discussion that led to me writing the comment like this, but I cannot figure out the original rationale. Reading the code as well, I would agree it doesn't look like anything changes for those cases.

Reply:
Let's remove this comment, then.